xmake/modules/detect/sdks/find_vstudio.lua

```diff
@@ -343,10 +343,18 @@ function _find_vstudio(opt)
                 table.insert(paths, path.join(logical_drive, "Program Files", "Microsoft Visual Studio " .. version, "VC"))
                 if version == "6.0" then
                     table.insert(paths, path.join(logical_drive, "Program Files", "Microsoft Visual Studio", "VC98", "Bin"))
+                    table.insert(paths, path.join(logical_drive, "Program Files (x86)", "Microsoft Visual Studio", "VC98", "Bin"))
                 end
             end
             vcvarsall = find_file("vcvarsall.bat", paths) or find_file("vcvars32.bat", paths)
         end
+        if not vcvarsall and version == "6.0" then
+            local paths = {
+                "$(reg HKEY_LOCAL_MACHINE\\SOFTWARE\\WOW6432Node\\Microsoft\\DevStudio\\6.0\\Products\\Microsoft Visual C++;ProductDir)\\Bin",
+                "$(reg HKEY_LOCAL_MACHINE\\SOFTWARE\\WOW6432Node\\Microsoft\\VisualStudio\\6.0\\Setup\\Microsoft Visual C++;ProductDir)\\Bin"
+            }
+            vcvarsall = find_file("vcvars32.bat", paths)
```

Comment:
we should add these paths in https://github.com/xmake-io/xmake/blob/cb8554de6bc050bc325c0eb3e146409d5aae819d/xmake/modules/detect/sdks/find_vstudio.lua#L320

Reply:
and it should be only for x64

Reply:
WOW6432Node  only for x64

Reply:
> WOW6432Node only for x64

so we should add `if os.arch() == "x64"`